examples/Makefile.am

```diff
@@ -44,23 +44,25 @@ VPP_INPUT_SOURCES = \
 
 AM_CPPFLAGS = $(AM_CFLAGS)
 YAMI_COMMON_LIBS = \
-	$(top_builddir)/common/libyami_common.la \
-	$(top_builddir)/vaapi/libyami_vaapi.la \
+	$(top_builddir)/libyami.la \
 	$(NULL)
 
 if ENABLE_X11
 YAMI_COMMON_LIBS += $(LIBVA_X11_LIBS) -lX11
 endif
 
+if BUILD_STATIC
+YAMI_COMMON_LIBS   += $(LIBVA_DRM_LIBS) $(LIBVA_X11_LIBS) -lX11
```

Comment:
$(LIBVA_DRM_LIBS) should go into YAMI_COMMON_LIBS unconditionally.  That is, it does not depend on whether we BUILD_STATIC or not.  Also, $(LIBVA_X11_LIBS) and -lX11 are already captured in the ENABLE_X11 conditional appropriately.  The only thing that should go in a BUILD_STATIC conditional is the LDFLAGS for our internal libs (i.e. libyami.a) to force the linker to capture them as a "whole archive".  That is, it prevents the Ubuntu version of GCC linker from stripping out our static initializers for the factories and such.